### DISCUSSION TITLE
USB: not so universal.

### DISCUSSION THREAD

**Discussion by peterhinch** [micropython/micropython#13418] [Using MicroPython] [answered]
I had cause to use an ESP8266 today. My fairly high-end Lenovo laptop wasn't having any of it.
```bash
$ ls /dev/tty*
```
indicated no device. Tried two different boards with the same result. On my older Dell laptop both worked fine, and I've used ESP8266 boards on other machines and never before encountered a problem.

By contrast an ESP32-S2 produces odd behaviour at the REPL on the Dell, but works fine on the Lenovo.

Have other people experienced similar differences between PC's?

**Comment by robert-hh** [comment]
I had a problem when installing Ubuntu on a old Laptop. There was the Braille reader service which had stolen /dev/ttyUSB0. I had to disable the related service, but do not recall which one. Google helped me. Using Fedora Linux at  the same machine I had no problem.

**Comment by GitHubsSilverBullet** [comment]
> I had cause to use an ESP8266 today. My fairly high-end Lenovo laptop wasn't having any of it.
> Have other people experienced similar differences between PC's?

Would be interesting to see the dmesg output  when plugging and unplugging the device.

**Reply by rkompass** [reply]
A few years ago there were many esp8266 boards with fake FTDI USB interface chips around. FTDI managed to distribute drivers that disabled these entirely. I had to throw quite a few esp8266 boards into the bin and recently ordered a pair of boards from adafruit for having a working esp8266.
What type of USB interface chip is on your board?

**Comment by robert-hh** [comment, marked as answer]
@peterhinch That was the post that worked for me: https://askubuntu.com/questions/1403705/dev-ttyusb0-not-present-in-ubuntu-22-04

**Reply by peterhinch** [reply]
Thank you. For anyone else with the problem do
```bash
$ sudo apt remove brltty
```
Odd that it only affects ESP8266. I've been using Pyboards, ESP32 variants and RP2 wit no issue.

**Reply by robert-hh** [reply]
As far as I could tell, it affected boards with a CH34x like the Wemos D1 Mini, but not boards with a CP210x like the Adafruit Feather ESP8266.

**Reply by scruss** [reply]
brltty — which occasionally insists on reinstalling itself on some Debian-based systems — only stomps on ttyUSBx. Pyboards, ESP32 variants and RP2 tend to be ttyACMx devices.

**Reply by jimmo** [reply]
Just in case someone finds this thread in the future: There was a similar problem in the past with Ubuntu (and likely other distros) installing and running `modemmanager` by default which would take over some `/dev/tty*` devices (I think only ttyACM from memory). `sudo apt remove modemmanager`
